src/components/Avatar/Avatar.jsx

```diff
@@ -40,25 +27,39 @@ export default class Avatar extends React.Component {
   }
 
   render () {
-    const { size, image, upload, onChange } = this.props
+    const { size, image, upload, onChange, username } = this.props
     const style = { backgroundImage: 'url(' + image + ')' }
 
-    if (upload) {
-      return (
-        <div className='upload'>
-          <input name='avatar' type='file' ref='fileDialog'
-            style={{ display: 'none' }} onChange={onChange}
-            accept='.jpg, .jpeg, .png'
-          />
+    return (
+      <div
+        style={{ width: size, height: size }}
+        onClick={upload ? this.openDialog : null}
+        className={classNames(
+          styles.Avatar,
+          { [styles.upload]: upload && (image || username) },
+          { [styles.new]: !image && !username },
+          { [styles.mobile]: size <= 32 }
+        )}
+      >
+        <input name='avatar' type='file' ref='fileDialog'
+          style={{ display: 'none' }} onChange={onChange}
+          accept='.jpg, .jpeg, .png'
+        />
+
+        <Camera className={styles.camera} size={42} />
 
-          {image
-            ? <Replace size={size} image={image} onClick={this.openDialog} />
-            : <NewUpload style={style} onClick={this.openDialog} />
-          }
-        </div>)
-    }
+        {!image && username && (
```

Comment:
same here
(Isn't there supposed to always have a username that is not undefined ?)